metacluster/include/metacluster/ListTenants.actor.h

```diff
@@ -111,15 +115,16 @@ Future<std::vector<std::pair<TenantName, MetaclusterTenantMapEntry>>> listTenant
     TenantName end,
     int limit,
     int offset = 0,
-    std::vector<TenantState> filters = std::vector<TenantState>()) {
+    std::vector<TenantState> filters = std::vector<TenantState>(),
+    TenantGroupName tenantGroup = ""_sr) {
 	state Reference<typename DB::TransactionT> tr = db->createTransaction();
 	state std::vector<std::pair<TenantName, MetaclusterTenantMapEntry>> results;
 
 	loop {
 		try {
 			tr->setOption(FDBTransactionOptions::READ_SYSTEM_KEYS);
 			tr->setOption(FDBTransactionOptions::READ_LOCK_AWARE);
-			if (filters.empty()) {
+			if (filters.empty() && tenantGroup.empty()) {
```

Comment:
Even if tenant group is set, I think you want to run the logic in this block. You would replace the call to `listTenantsTransaction` with something like `listTenantGroupTenantsTransaction`.